pkg/util/net/ip_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package net
+
+import (
+	"net/netip"
+	"reflect"
+	"testing"
+)
+
+func TestIPsSplitV4V6(t *testing.T) {
+	tests := []struct {
+		name string
+		ips  []string
+		ipv4 []string
+		ipv6 []string
+	}{
+		{
+			name: "include valid and invalid ip addresses",
+			ips:  []string{"1.0.0.0", "256.255.255.255", "localhost", "::1", "2001:db8:::1"},
+			ipv4: []string{"1.0.0.0"},
+			ipv6: []string{"::1"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if ipv4, ipv6 := IPsSplitV4V6(tt.ips); !reflect.DeepEqual(ipv4, tt.ipv4) || !reflect.DeepEqual(ipv6, tt.ipv6) {
+				t.Errorf("IPsSplitV4V6() got %v %v, want %v %v", ipv4, ipv6, tt.ipv4, tt.ipv6)
+			}
+		})
+	}
+}
+
+func TestParseIPsSplitToV4V6(t *testing.T) {
```

Comment:
May be better written as multiple test cases

Reply:
Thanks, I have added multiple test cases